tools/pipelines/build-docs-website-deploy.yml

```
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# build-docs-website-deploy pipeline
+
+name: $(Build.BuildId)
+
+schedules:
+  - cron: "0 4 * * *"
+    displayName: Daily 8pm (PST) website deploy
+    branches:
+      include:
+        - main
+    always: true
+
+trigger: none
+pr: none
+
+parameters:
+- name: deployOverride
+  displayName: Deployment Override (default = based on branch)
+  type: string
+  default: default
+  values:
+    - default
+    - skip
+    - force
+- name: deployEnvironment
+  displayName: Static web app environment to deploy to
+  type: string
+  default: new
+  values:
+    - new
+    - old
+
+variables:
+- name: isMain
+  value: ${{ eq(variables['Build.SourceBranchName'], 'main') }}
+- name: shouldDeploy
+  value: ${{ or(
+    eq(parameters.deployOverride, 'force'),
+    and(eq(variables.isMain, true), eq(parameters.deployOverride, 'default'))
+    )}}
```

In this new file `isMain` has a single use so I'd lean towards inlining it:
```suggestion
- name: shouldDeploy
  value: ${{ or(
    eq(parameters.deployOverride, 'force'),
    and(eq(variables['Build.SourceBranchName'], 'main'), eq(parameters.deployOverride, 'default'))
    )}}
```